src/proto/streams/recv.rs

```diff
@@ -599,6 +590,22 @@ impl Recv {
         Ok(())
     }
 
+    pub fn ignore_data(&mut self, sz: WindowSize) -> Result<(), RecvError> {
+        // Ensure that there is enough capacity on the connection...
+        self.consume_connection_window(sz)?;
```

Comment:
Is the thought that this window underflow is important to surface, even in cases when we are going to totally ignore the frame?  I haven't thought deeply, but if we can't trust the client to understand the state of these streams, might it better to just ignore the window in this case?

Reply:
This is being lenient over whether the peer has received our `RST_STREAM` frame yet. I'd still expect a peer to respect the connection flow control.